api/osf_groups/views.py

```diff
@@ -0,0 +1,224 @@
+from django.apps import apps
+from django.db.models import Q
+
+from rest_framework import generics, permissions as drf_permissions, exceptions
+from rest_framework.exceptions import NotFound, ValidationError
```

Comment:
Nit: No need to import `exceptions` above if you're going to import individual exceptions. I'm fine with either way, just needs to be consistent.